lib/buffer.js

```diff
@@ -133,13 +133,20 @@ Buffer.from = function(value, encodingOrOffset, length) {
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+function assertSize(size) {
+  if (typeof size !== 'number') {
+    const err = new TypeError('"size" argument must be a number');
+    Error.captureStackTrace(err, assertSize);
```

Comment:
Yep. Just hiding internal impl details. I'm fine with pulling that back out
if it's problematic
On Apr 11, 2016 4:05 PM, "Trevor Norris" <notifications@github.com> wrote:

> In lib/buffer.js
> <https://github.com/nodejs/node/pull/5833#discussion_r59297009>:
>
> > @@ -133,13 +133,20 @@ Buffer.from = function(value, encodingOrOffset, length) {
> >  Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
> >  Object.setPrototypeOf(Buffer, Uint8Array);
> >
> > +function assertSize(size) {
> > +  if (typeof size !== 'number') {
> > +    const err = new TypeError('"size" argument must be a number');
> > +    Error.captureStackTrace(err, assertSize);
>
> I'm confused. Is the only reason for this to remove assertSize() from the
> stack?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/5833/files/54c5a36554f6a96ad4f2cb8dc415611b3b853fd7#r59297009>
>